src/Strategy/Scala/SbtDependencyTreeJson.hs

```diff
@@ -0,0 +1,148 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Strategy.Scala.SbtDependencyTreeJson (
+  analyze,
+  parseSbtArtifact,
+) where
+
+import Control.Effect.Diagnostics (Diagnostics)
+import Control.Monad (void)
+import Control.Monad.Identity (Identity)
+import Data.Aeson (
+  FromJSON (parseJSON),
+  withArray,
+  withObject,
+  (.:),
+ )
+import Data.Foldable (for_)
+import Data.Maybe (fromMaybe)
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Vector (toList)
+import Data.Void (Void)
+import DepTypes (
+  DepType (MavenType),
+  Dependency (..),
+  VerConstraint (CEq),
+ )
+import Effect.Grapher (GrapherC, Has, deep, direct, edge, evalGrapher, run)
+import Effect.ReadFS (ReadFS, readContentsJson)
+import Graphing (Graphing, shrinkRoots)
+import Path (Abs, File, Path)
+import Strategy.Scala.Common (SbtArtifact (..))
+import Text.Megaparsec (
+  MonadParsec (try),
+  Parsec,
+  empty,
+  errorBundlePretty,
+  optional,
+  runParser,
+  some,
+  (<|>),
+ )
+import Text.Megaparsec.Char (alphaNumChar, char)
+import Text.Megaparsec.Char.Lexer qualified as Lexer
+
+-- | Represents output of Sbt's Treeview command.
+-- Ref: https://github.com/sbt/sbt/blob/master/main/src/main/scala/sbt/internal/graph/rendering/TreeView.scala#L22
+--
+-- It looks like:
+-- > [
+-- >   {
+-- >     "text": "groupId:artifactId:version",
+-- >     "children": [
+-- >       {
+-- >         "text": "...",
+-- >         "children": [
+-- >           ...
+-- >         ]
+-- >       },
+-- >     ]
+-- >   }
+-- > ]
+-- -
+newtype SbtTree = SbtTree [SbtDep]
+  deriving (Eq, Ord, Show)
+
+data SbtDep = SbtDep
+  { artifact :: SbtArtifact
+  , dependsOn :: [SbtDep]
+  }
+  deriving (Show, Eq, Ord)
+
+instance FromJSON SbtTree where
+  parseJSON = withArray "SbtDepRoot" $ \arr -> do
+    SbtTree . toList <$> traverse parseJSON arr
+
+instance FromJSON SbtDep where
+  parseJSON = withObject "SbtDep" $ \obj -> do
+    txt <- obj .: "text"
+    children <- obj .: "children"
+
+    artifact <- case runParser parseSbtArtifact "sbtArtifact" txt of
+      Left err -> fail $ errorBundlePretty err
+      Right sa -> pure sa
+    pure $ SbtDep artifact children
+
+type ParserT = Parsec Void Text
+
+parseSbtArtifact :: ParserT SbtArtifact
+parseSbtArtifact = do
+  groupId <- parseValidProjectIdentifier
+  artifactId <- (":" >> parseValidProjectIdentifier)
+  version <- lexeme (":" >> parseValidProjectIdentifier)
+  evictedByVersion <- try . optional $ parseEviction
+  pure $
+    SbtArtifact
+      groupId
+      artifactId
+      (fromMaybe version evictedByVersion)
+  where
+    sc :: ParserT ()
+    sc = Lexer.space (void $ some $ char ' ' <|> char '\t') empty empty
+
+    lexeme :: ParserT a -> ParserT a
+    lexeme = Lexer.lexeme sc
+
+    symbol :: Text -> ParserT Text
+    symbol = Lexer.symbol sc
+
+    parseEviction :: ParserT (Text)
+    parseEviction = lexeme ("(evicted by" <|> "(evicted by") *> parseValidProjectIdentifier <* symbol ")"
```

Comment:
Maybe I just don't understand what's going on here, but having `"(evicted by"` twice in a row within parens looks wrong to me. Is that correct?

Reply:
Also, from the specs it looks like you expect a line like "a:b.c:1.0 (evicted by 2.0)" to get parsed as having a version of 2.0. Can you add a comment explaining why that is?